extensions/attention-stream/src/youtube/utils/traverseFishing.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Recrusive tree traversal of JSON object:
+ * Returns or deletes one or multiple keys
+ * depending on parameters
+ */
+const traverse = (
+  jsonObject,
+  shouldDelete,
+  returnFirst,
+  keys,
+  keepKeys = false
+) => {
+  let results = [];
+
+  const traverseHelper = (jsonObject) => {
+    for (var key in jsonObject) {
+      if (keys.includes(key)) {
+        if (shouldDelete) {
+          delete jsonObject[key];
+        } else {
+          if (keepKeys) {
+            results.push([key, jsonObject[key]]);
+          } else {
+            results.push(jsonObject[key]);
+          }
+        }
+
+        if (returnFirst) {
+          break;
+        }
+      }
+      if (jsonObject[key] !== null && typeof jsonObject[key] == "object") {
+        // Recurse
+        traverseHelper(jsonObject[key]);
+      }
+    }
+  };
+
+  traverseHelper(jsonObject);
```

Comment:
What happens if we hit the max recursion depth for JS? Is there a way to check upfront that this can't happen?

Reply:
Oh I hadn't thought about that. It looks like max recursion depth these days is 9000 at a minimum. Since these objects are essentially used by the YouTube front-end as a data store, I think there's a 0% chance they have a depth of more than say, 25.